pallets/pips/src/lib.rs

```diff
@@ -528,95 +568,126 @@ decl_module! {
         #[weight = (150_000_000, DispatchClass::Operational, Pays::Yes)]
         pub fn set_proposal_cool_off_period(origin, duration: T::BlockNumber) {
             T::CommitteeOrigin::ensure_origin(origin)?;
-            Self::deposit_event(RawEvent::ProposalDurationChanged(SystematicIssuers::Committee.as_id(), Self::proposal_cool_off_period(), duration));
+            Self::deposit_event(RawEvent::ProposalCoolOffPeriodChanged(SystematicIssuers::Committee.as_id(), Self::proposal_cool_off_period(), duration));
             <ProposalCoolOffPeriod<T>>::put(duration);
         }
 
         /// Change the default enact period.
-        #[weight = (300_000_000, DispatchClass::Operational, Pays::Yes)]
+        #[weight = (150_000_000, DispatchClass::Operational, Pays::Yes)]
         pub fn set_default_enactment_period(origin, duration: T::BlockNumber) {
             T::CommitteeOrigin::ensure_origin(origin)?;
-            let previous_duration = <DefaultEnactmentPeriod<T>>::get();
+            let prev = <DefaultEnactmentPeriod<T>>::get();
             <DefaultEnactmentPeriod<T>>::put(duration);
-            Self::deposit_event(RawEvent::DefaultEnactmentPeriodChanged(SystematicIssuers::Committee.as_id(), duration, previous_duration));
+            Self::deposit_event(RawEvent::DefaultEnactmentPeriodChanged(SystematicIssuers::Committee.as_id(), prev, duration));
+        }
+
+        /// Change the maximum skip count (`max_pip_skip_count`).
+        /// New values only
+        #[weight = (150_000_000, DispatchClass::Operational, Pays::Yes)]
+        pub fn set_max_pip_skip_count(origin, new_max: SkippedCount) {
+            T::CommitteeOrigin::ensure_origin(origin)?;
+            let prev_max = MaxPipSkipCount::get();
+            MaxPipSkipCount::put(new_max);
+            Self::deposit_event(RawEvent::MaxPipSkipCountChanged(SystematicIssuers::Committee.as_id(), prev_max, new_max));
+        }
+
+        /// Change the maximum number of active PIPs before community members cannot propose anything.
+        #[weight = (150_000_000, DispatchClass::Operational, Pays::Yes)]
+        pub fn set_active_pip_limit(origin, new_max: u32) {
+            T::CommitteeOrigin::ensure_origin(origin)?;
```

Comment:
Do we need to check the new limit is above the current number of active Pips?

Reply:
Not necessarily. If you set it below the current, then all that happens is no more community PIPs can be made until some are not active anymore / limit is raised, which is ostensibly a useful hard-freeze functionality.

Reply:
Makes sense.